UP-19-ds-replace-mongo-db-with-sqlite-db

somersgreg opened the pull request:
## Description

- Moved all Mongo files to `mongo_files` directory

- Implemented SQL using SQLAlchemy:  
  - `main.py`
    - Houses FastAPI endpoints
  - `table_models.py`
    - Contains the blueprint for our SQL tables
  - `database.py`
    - Establishes our foundation for connecting to the database
  - `schema.py`
    - Uses Pydantic to validate data
  - `services.py`
    - Contains the component-based CRUD operations that our API uses 

- Niger wanted us to practice SQL implementation as its an industry standard 

## Type of change

- [x] New feature
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows PEP8 style guide
- [ ] I have removed unnecessary print statements from my code
- [ ] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [ ] No commented-out code
- [ ] Size of pull request kept to a minimum
- [ ] Pull request description clearly describes changes made & motivations for said changes

## Loom
[Loom Video](https://www.loom.com/share/db2f3ad3974c45f990f4ddc9585bfabd)

somersgreg committed:
Moved Mongo, implemented SQL

nlittlepoole commented on app/database.py:
```diff
@@ -1,73 +1,13 @@
+load_dotenv()
+Base = declarative_base()
+engine = create_engine(
+    url=os.getenv("SQL_DB_URL"),
```
`url=os.getenv("SQL_DB_URL", default="sqlite:///:memory:")`

is better here as the default value is what gets returned if the user forgets to set the `SQL_DB_URL` environment variable. `sqlite:///:memory:` means the database will be stored in memory rather than on the harddrive.

nlittlepoole reviewed (changes requested):
Everything else looks good, just want to handle the default

somersgreg committed:
Update database.py

- returns default value if the user forgets to set the SQL_DB_URL environment variable
    - sqlite:///:memory: means the database will be stored in memory rather than on the harddrive.

jinjahninjah commented on run.sh:
```diff
@@ -1 +1,2 @@
-python -m uvicorn app.api:API --reload
+python -m uvicorn app.main:app --reload
+#uvicorn main:app --reload
```
The only thing I have is to remove this commented out code here real quick and you should be good to go 👍

somersgreg committed:
UP-19

 - Remove commented code in run.sh

jinjahninjah reviewed (approved):
Looks great! Excited to see how the app progresses as now we get to have some fun.

jinjahninjah reviewed (commented):
This was a big undertaking and will help cement the foundation of SQLAlchemy and SQLite 👍

nlittlepoole reviewed (approved):
📈
